test/e2e/shared/aws.go

```diff
@@ -771,7 +792,7 @@ func GetSubnet(e2eCtx *E2EContext, subnetID string) (*ec2.Subnet, error) {
 	return result.Subnets[0], nil
 }
 
-func CreateSubnet(e2eCtx *E2EContext, subnetName string, cidrBlock string, az string, vpcID string) (*ec2.Subnet, error) {
+func CreateSubnet(e2eCtx *E2EContext, clusterName string, cidrBlock string, az string, vpcID string, st string) (*ec2.Subnet, error) {
```

Comment:
We will need  a way to identify if a subnet is public or not to attach proper gateway rules to route table when this method is used to create the subnet. 
In the code, we do this using `isPublic` field.

Reply:
Tags could be used.